### PR TITLE
menu: Pass disabled mod to every nested menu-item

### LINKED ISSUE #352: menu: pass disabled mod to every nested menu-item

### PR DESCRIPTION
**Pull request by sipayRT**
Close #352

### CODE REVIEW

**Comment by dfilatov** [comment]
@sipayRT you should do that in js as well

**Comment by sipayRT** [comment]
@dfilatov check, plz

File: common.blocks/menu/menu.js [hunk 1]
```diff
@@ -37,6 +37,14 @@ provide(BEMDOM.decl({ block : this.name, baseBlock : BaseControl }, /** @lends m
                     .__base.apply(this, arguments);
                 this._hoveredItem && this._hoveredItem.delMod('hovered');
             }
+        },
+
+        'disabled' : {
```
Comment (dfilatov) [inline common.blocks/menu/menu.js]: You can reduce nesting:

``` js
'disabled' : function(modName, modVal) {
    this.getItems().forEach(function(menuItem){
        menuItem.setMod(modName, modVal);
    });
}
```

File: common.blocks/menu/menu.bh.js [hunk 2]
```diff
@@ -4,6 +4,7 @@ module.exports = function(bh) {
         ctx
             .js(true)
             .tParam('_menuTheme', ctx.mod('theme'))
+            .tParam('_menuDisabled', ctx.mod('disabled'))
```
Comment (dfilatov) [inline common.blocks/menu/menu.bh.js]: Might be we'll pass `_menuMods` instead of passing each modifier separately?

**Comment by sipayRT** [comment]
@dfilatov fixed and rebased

**Comment by dfilatov** [comment]
@sipayRT we need to add specs for js behaviour

**Comment by sipayRT** [comment]
@dfilatov thanks for reminding me - there was an error in BEMHTML template

File: common.blocks/menu/menu.spec.js [hunk 3]
```diff
@@ -49,6 +49,13 @@ describe('menu', function() {
             menu.delMod('disabled');
             control.attr('tabindex').should.be.equal('0');
         });
+
+        it('should pass disabled mod to menu items', function() {
+            menu.setMod('disabled');
+            menuItems.forEach(function(menuItem) {
+                menuItem.hasMod('disabled').should.be.true;
+            });
+        });
```
Comment (dfilatov) [inline common.blocks/menu/menu.spec.js:62]: Add a test case for the opposite situation -- all `menu-item` should be enabled while `menu` is being enabled

Comment (dfilatov) [inline common.blocks/menu/menu.spec.js:62]: @sipayRT ping

**Comment by sipayRT** [comment]
@dfilatov sorry for the delay - we had holidays. Specs was added

**Comment by dfilatov** [comment]
:shipit:

**Pull request merged**